### ISSUE TITLE
libvirt: the default virtual network cannot be removed

### ISSUE THREAD

**Issue by alexbakker** [NixOS/nixpkgs#73418]
**Describe the bug**
The default virtual network for libvirt cannot be persistently removed or disabled.

**To Reproduce**
1. Enable libvirtd with ``virtualisation.libvirt.enable = true;``.
2. Verify that the default network exists with: ``virsh net-list``.
3. Remove the 'default' network with ``virsh net-destroy default && virsh net-undefine default``.
4. Verify that the default network has been removed with: ``virsh net-list``.
5. Restart libvirtd: ``systemctl restart libvirtd``.
6. Notice that the default network has reappeared.

**Expected behavior**
I expected the 'default' network to still be gone after a restart of libvirtd. 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.10, NixOS, 19.09.1223.cb2cdab7136 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"nixos-19.09.1223.cb2cdab7136, home-manager-19.09"`
 - channels(alex): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: libvirt
# a list of nixos modules affected by the problem
module: libvirtd
```

**Comment by volth** [comment]
I guess, network changes which are suppose to survive daemon restart or reboot, should be made declaratively.

Probably in `virtualisation.libvirtd.extraConfig` or `virtualisation.libvirtd.extraOptions`
Otherwise a new option should be introduced.

**Comment by volth** [comment]
changes made in brides using `brctl` are restored on `libvirtd` restart too

**Comment by alexbakker** [comment]
This is still an issue.

**Comment by euank** [comment]
Unstale; I'll try to find the time to poke at this since I've been doing a little work on the libvirt package.

**Comment by Xpl0itR** [comment]
Still an issue.

**Comment by dasJ** [comment]
Caused by this: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/virtualisation/libvirtd.nix#L194

**Comment by dasJ** [comment]
still annoying to me

**Comment by fpletz** [comment]
Fixed by @amarshall in #186358. Thanks!

**Comment by rickykresslein** [comment]
I've stumbled upon this when trying to figure out how to set a default network. Now that it is not set automatically, how can I set it?
